lab_5_scrapper/scrapper.py

```diff
@@ -1,82 +1,199 @@
+
+        if not all(re.match('https?://w?w?w?.zebra-tv.ru/', url) for url in seed_urls):
+            raise IncorrectSeedURLError('seed URL does not match the standard pattern \
+                                        or does not correspond to the target website')
+
+        if total_articles not in range(1, 151):
+            raise NumberOfArticlesOutOfRangeError('total number of articles is \
+                                                  out of range from 1 to 150')
+
+        if not isinstance(total_articles, int) or isinstance(total_articles, bool):
+            raise IncorrectNumberOfArticlesError('total number of articles to parse is not integer')
+
+        if not isinstance(headers, dict):
+            raise IncorrectHeadersError('headers are not in a form of dictionary')
+
+        if not isinstance(encoding, str):
+            raise IncorrectEncodingError('encoding must be specified as a string')
+
+        if timeout not in range(1, 60):
+            raise IncorrectTimeoutError('timeout value must be a positive integer less than 60')
+
+        if not isinstance(verify_cert, bool):
+            raise IncorrectVerifyError('verify certificate value must either be True or False')
 
     def get_seed_urls(self) -> list[str]:
         """
         Retrieve seed urls
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            seed_urls = json.load(f)['seed_urls']
+
+        return list(str(url) for url in seed_urls)
 
     def get_num_articles(self) -> int:
         """
         Retrieve total number of articles to scrape
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            total_articles = json.load(f)['total_articles_to_find_and_parse']
+
+        return int(total_articles)
 
     def get_headers(self) -> dict[str, str]:
         """
         Retrieve headers to use during requesting
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            headers = json.load(f)['headers']
+
+        pairs = [(str(key), str(val)) for key, val in headers.items()]
+        return dict(pairs)
 
     def get_encoding(self) -> str:
         """
         Retrieve encoding to use during parsing
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            encoding = json.load(f)['encoding']
+
+        return str(encoding)
 
     def get_timeout(self) -> int:
         """
         Retrieve number of seconds to wait for response
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            timeout = json.load(f)['timeout']
+
+        return int(timeout)
 
     def get_verify_certificate(self) -> bool:
         """
         Retrieve whether to verify certificate
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
```

Comment:
come-on, do not read json for each access of attribute - this is enormously expensive. it should be done once on init of object, when you load json, read, create DTO and then assign to self.ATTR_NAME appropriate attributes. getters are needed for outer world to access your internal attributes.